src/lib/withBaseURL.ts

```diff
@@ -19,3 +19,15 @@ export function withWebSocketURL(querystring: string): string {
   }
   return url + querystring
 }
+
+export function withMlephantWebSocketURL(querystring: string): string {
+  const url = env().VITE_MLEPHANT_WEBSOCKET_URL
+  if (!url) {
+    return ''
+  }
+  // Append to existing query string if it exists
+  if (url.includes('?') && querystring.startsWith('?')) {
+    return url + querystring.replace(/^\?/, '&')
+  }
+  return url + querystring
```

Comment:
I guess the original util function could cover just that part and we could have the new with* functions calling it, after the env() call?